ibm/service/kubernetes/resource_ibm_container_addons.go

```diff
@@ -528,12 +530,27 @@ func flattenAddOns(result []v1.AddOn) (resp *schema.Set, err error) {
 }
 func resourceIBMContainerAddOnsUpdate(d *schema.ResourceData, meta interface{}) error {
 	manageAllAddons := d.Get("manage_all_addons").(bool)
+	managed_addons := d.Get("managed_addons").([]interface{})
 	var addOnName []string
+	// If the user wants terraform to manage the defined addons
 	if !manageAllAddons {
+		// Case 1 - User has requested to manage addons after terraform creation and initialisation, but there is no change in the addon configuration
+		if len(managed_addons) == 0 {
+			if !d.HasChange("addons") {
+				var definedAddons []string
```

Comment:
As discussed, we can reuse the existing addOnName variable here and get rid of Get & Set function and follow the existing flow.